### DISCUSSION TITLE
How to get size of master ( or current window ) in loop function

### DISCUSSION THREAD

**Discussion by manuel-koch** [AllenDang/giu#578] [Q&A] [answered]
How can I get current size of the master window within my `loop()` function ?

I tried the following
- giu.GetAvailableRegion()
- imgui.ContentRegionAvail()
- imgui.WindowSize()
but none actually returned the size I initially used for
`wnd := giu.NewMasterWindow("Hello world", 200, 200, 0)`
nor an updated size when I resized the window with my mouse.

**Comment by manuel-koch** [comment]
I think it was just a misunderstanding of imgui / giu on my part.

The following seems to work for a simple loop function content:
```
giu.SingleWindow().Layout(
   giu.Custom(func() {
      w, h := giu.GetAvailableRegion()
   }),
   giu.Label("Hello"),
)
```

The trick is to evaluate the size of the current window *within* the window code, not outside.

**Reply by gucio321** [reply]
You can get window size by
wnd:= giu.Window("wnd")
w, h := wnd.CurrentSize()
wnd.Layout(....)

**Comment by AllenDang** [comment, marked as answer]
@manuel-koch MasterWindow has a method named "GetSize".
```go
wnd := giu.NewMasterWindow("Hello world", 200, 200, 0)
width, height := wnd.GetSize()
```
